dayrepo/snippets/views.py

```diff
@@ -225,117 +226,179 @@ def get(self, request, snippet_id):
-                snippet_id=post_trouble.snippet_id,
-                trouble_situation=req.get("trouble_situation"),
-                trouble_cause=req.get("trouble_cause"),
-                trouble_support=req.get("trouble_support"),
-            )
-            post_trouble.delete()
-            duties_trouble.save()
-            
-            
-            # 工程テーブルフォーム保存
-            form_process_count = len(req.getlist("via_point"))
-            for i in range(form_process_count):
-                process = Process(
-                    id=post_process[i].id,
-                    snippet_id=post_process[i].snippet_id,
-                    start_time=req.getlist("start_time")[i + 1],
-                    end_time=req.getlist("end_time")[i + 1],
-                    start_point=req.getlist("start_point")[i + 1],
-                    end_point=req.getlist("end_point")[i + 1],
-                    via_point=req.getlist("via_point")[i],
-                    client=req.getlist("client")[i],
-                    goods=req.getlist("goods")[i],
-                    load_situation=req.getlist("load_situation")[i],
-                    load_mileage=req.getlist("load_mileage")[i],
-                    hollow_mileage=req.getlist("hollow_mileage")[i],
-                    is_load_situation=req.getlist("is_load_situation")[i],
-                )
-                if process.is_load_situation == "on":
-                    process.is_load_situation = True
-                else:
-                    process.is_load_situation = False
-                post_process[i].delete()
-                process.save()
-            post_snippet=snippet
-            # トップ画面へ
-            return redirect(to="snippet_list")
+                process.is_load_situation = False
+            post_process[i].delete()
+            process.save()
+        post_snippet = snippet
+        # トップ画面へ
+        return redirect(to="snippet_list")
+
 snippet_edit = SnippetEditView.as_view()
 
+
+class DbDeletePattern():
+    def snippet_del(self,target_id:int):
+        # snippetと紐づいたdutiestroubleの削除
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        # snippetと紐づいたprocessの削除
+        process = get_list_or_404(
+            Process, snippet_id=target_id
+        )
+        for process_object in process:
+            process_object.delete()
+
+        snippet = Snippet.objects.get(
+            pk=target_id
+        )
+        snippet.delete()
+
+
+        checklist = Checklist.objects.get(
+            pk=snippet.checklist_id.id
+        )
+
+        # 対象のsnippetが削除されたことを反映し、
+        # checklistのみの投稿として表示、判定させる
+        checklist.is_snippet_make = False
+        checklist.save()
+
+    # 未提出postdataのchecklistを削除する
+    def checklist_del(self,target_id:int):
+        checklist = Checklist.objects.get(
+        pk=target_id
+        )
+        checklist.delete()
+
+    # 提出済みpostdataをすべて削除する
```

Comment:
このメソッド的には、　post かどうかには関心がないと思う〜
post に関する情報をここに付与してしまうと、それは依存関係が出来上がっていることにもなるので、いらないと思います。
```suggestion
    # 提出済み data をすべて削除する
```